src/service-providers/mailchimp/ProviderSidebar.js

```diff
@@ -2,13 +2,81 @@
  * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
-import { Fragment, useEffect } from '@wordpress/element';
-import { ExternalLink, SelectControl, Spinner, Notice } from '@wordpress/components';
+import { Fragment, useEffect, useState } from '@wordpress/element';
+import {
+	ExternalLink,
+	SelectControl,
+	Spinner,
+	Notice,
+	FormTokenField,
+} from '@wordpress/components';
 
 /**
  * Internal dependencies
  */
-import { getListInterestsSettings } from './utils';
+import { getListInterestsSettings, getListTags, getTagNames, getTagIds } from './utils';
+
+const SegmentsSelection = ( {
+	onUpdate,
+	inFlight,
+	chosenInterestId,
+	availableInterests,
+	chosenTags,
+	availableTags,
+} ) => {
+	const [ segmentsData, setSegmentsData ] = useState( {
+		interest_id: chosenInterestId,
+		tag_ids: chosenTags,
+	} );
+	const updateSegmentsData = data => setSegmentsData( { ...segmentsData, ...data } );
+
+	// Update with real data after local (optimistic) update.
+	useEffect(() => {
+		setSegmentsData( {
+			interest_id: chosenInterestId,
+			tag_ids: chosenTags,
+		} );
+	}, [ chosenInterestId, chosenTags.join() ]);
+
+	const [ isInitial, setIsInitial ] = useState( true );
+	useEffect(() => {
+		if ( ! isInitial ) {
+			onUpdate( segmentsData );
+		}
+		setIsInitial( false );
+	}, [ segmentsData.interest_id, segmentsData.tag_ids.join() ]);
+
+	return (
+		<Fragment>
+			{ availableInterests.length && (
+				<SelectControl
+					label={ __( 'Groups', 'newspack-newsletters' ) }
+					value={ segmentsData.interest_id }
+					options={ [
+						{
+							label: __( '-- Select a group --', 'newspack-newsletters' ),
+							value: 'no_interests',
+						},
+						...availableInterests,
+					] }
+					onChange={ interest_id => updateSegmentsData( { interest_id } ) }
+					disabled={ inFlight }
+				/>
+			) }
+			{ availableTags.length && (
+				<FormTokenField
```

Comment:
Not sure if there's anything that can be done about this, but when I first start typing in the field I get autocomplete UI that seems to be based on browser cache or something, not the list of tags from Mailchimp. In this screenshot, I have no tag named "sports" in Mailchimp:

<img width="290" alt="Screen Shot 2020-07-05 at 2 35 08 PM" src="https://user-images.githubusercontent.com/1477002/86539706-cb33d900-becc-11ea-9942-a49f47c7a2ec.png">

Reply:
Unfortunately the `id` of the rendered `input` is [not overridable](https://github.com/WordPress/gutenberg/blob/059e8bebbc19e67f2685d56dc000b4439eadbebb/packages/components/src/form-token-field/index.js#L694) :/

Reply:
Oh well, not big deal!